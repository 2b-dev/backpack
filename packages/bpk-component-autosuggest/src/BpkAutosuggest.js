import React from 'react'
import Autosuggest from 'react-autosuggest'

import './bpk-autosuggest.scss'

const BpkAutosuggest = (props) => <Autosuggest {...props} />

BpkAutosuggest.defaultProps = {
  theme: {
    container: 'bpk-autosuggest__container',
    containerOpen: 'bpk-autosuggest__container--open',
    input: 'bpk-autosuggest__input',
    suggestionsContainer: 'bpk-autosuggest__suggestions-container',
    suggestionsList: 'bpk-autosuggest__suggestions-list',
    suggestion: 'bpk-autosuggest__suggestion-item',
    suggestionFocused: 'bpk-autosuggest__suggestion-item--focused',
    sectionContainer: 'bpk-autosuggest__section-container',
    sectionTitle: 'bpk-autosuggest__section-title'
  }
}

export default BpkAutosuggest
