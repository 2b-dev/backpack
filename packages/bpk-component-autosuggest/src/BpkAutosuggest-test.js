import React from 'react'
import renderer from 'react/lib/ReactTestRenderer'
import BpkAutosuggest from './BpkAutosuggest'

describe('BpkAutosuggest', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<BpkAutosuggest
      suggestions={['Edinburgh', 'Glasgow', 'London']}
      value={'Edinburgh'}
      id={'snapshot'}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
