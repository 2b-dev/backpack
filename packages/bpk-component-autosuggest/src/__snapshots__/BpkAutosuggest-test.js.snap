exports[`BpkAutosuggest should render correctly 1`] = `
<div
  className="bpk-autosuggest__container">
  <input
    aria-activedescendant={null}
    aria-autocomplete="list"
    aria-expanded={false}
    aria-haspopup={false}
    aria-owns="react-autowhatever-snapshot"
    autoComplete="off"
    className="bpk-autosuggest__input"
    onBlur={[Function onBlur]}
    onChange={[Function onChange]}
    onFocus={[Function onFocus]}
    onKeyDown={[Function bound onKeyDown]}
    role="combobox"
    type="text"
    value="Edinburgh" />
  <div
    className="bpk-autosuggest__suggestions-container"
    id="react-autowhatever-snapshot" />
</div>
`;
